test/bigquery.ts

```diff
@@ -235,12 +244,12 @@ describe('BigQuery', () => {
       assert.deepStrictEqual(calledWith.scopes, expectedScopes);
     });
 
-    it('should not modify options argument', () => {
+    it('should not correctly modify options argument', () => {
```

Comment:
Does this test still work as intended? It was making sure the object the user gave didn't have any properties tampered with by the constructor.

Reply:
@stephenplusplus This is the issue I was referencing in a previous comment. The library originally ensured that no properties were mutated before being set, but if we are going to use baseUrl to set apiEndpoint, we have to then either assume the apiEndpoint will now be changed to include a protocol before being set (before it always expected the value to not include a protocol), or include a check for whether apiEndpoint has a protocol whenever we're using it to set a request uri in other places in the library. I Do you have an opinion on this?

Reply:
For example, [here](https://github.com/googleapis/nodejs-bigquery/blob/1e56383da5e6d8ce1335a711b32fea1155bddada/src/table.ts#L1417) is how we originally set the uri. If we are setting the apiEndpoint using provided baseUrl, I think we will have to assume apiEndpoint value includes the protocol.

Reply:
Gotcha, thanks. This test was taking itself a bit more literally, I think, and making sure that the object itself wasn't changed by our code-- i.e., if we took their config object, overwrote their properties / added new ones, etc. We just want to basically leave the user's input object alone by making sure we're working from a cloned object in the constructor.